rmf_site_editor/src/workspace.rs

```diff
@@ -184,31 +101,22 @@ impl Plugin for WorkspacePlugin {
     fn build(&self, app: &mut App) {
         app.add_event::<ChangeCurrentWorkspace>()
             .add_event::<CreateNewWorkspace>()
-            .add_event::<SaveWorkspace>()
-            .add_event::<SaveWorkcell>()
-            .add_event::<LoadWorkcell>()
             .init_resource::<CurrentWorkspace>()
             .init_resource::<RecallWorkspace>()
-            .init_resource::<SaveWorkspaceChannels>()
+            .init_resource::<FileDialogServices>()
             .init_resource::<WorkspaceLoadingServices>()
+            .init_resource::<WorkspaceSavingServices>()
             .add_systems(
                 Update,
-                (
-                    dispatch_new_workspace_events,
-                    sync_workspace_visibility,
-                    workspace_file_save_complete,
-                ),
+                (dispatch_new_workspace_events, sync_workspace_visibility),
             );
-        #[cfg(not(target_arch = "wasm32"))]
-        app.add_systems(Update, dispatch_save_workspace_events);
```

Comment:
There are several changes to this file, I'll just highlight them in this snippet that offers an overview.

Model saving now uses workflows, furthermore I separated the `rfd` file dialog spawning into a new `FileDialogServices` that provides workflows that allow users to pick files for loading / saving (optionally with filters for extensions), as well as picking folders for exporting (useful for SDF / URDF exporting).

Apps that need to handle workspaces will most likely need to create their own loading and saving services, however now they can reuse the `FileDialogServices`, as well as avoid having to worry about `rfd` so we can keep it as a private dependency.
Note that there are other parts in the codebase where we could use the new service rather than `rfd` (specifically three instances, saving lights, saving navgraphs and browsing local asset sources) however I left them out for now to try to limit the scope of this PR, I added TODOs in their place.